packages/parameters/src/AppConfigProvider.ts

```diff
@@ -0,0 +1,115 @@
+import { BaseProvider, DEFAULT_PROVIDERS } from './BaseProvider';
+import {
+  AppConfigDataClient,
+  StartConfigurationSessionCommand,
+  GetLatestConfigurationCommand,
+} from '@aws-sdk/client-appconfigdata';
+import type {
+  StartConfigurationSessionCommandInput,
+  GetLatestConfigurationCommandInput,
+} from '@aws-sdk/client-appconfigdata';
+import type { AppConfigGetOptionsInterface } from './types/AppConfigProvider';
+
+class AppConfigProvider extends BaseProvider {
+  public client: AppConfigDataClient;
+  private application: string;
+  private environment: string;
+  private latestConfiguration: Uint8Array | undefined;
+  private token: string | undefined;
+
+  /**
+   * It initializes the AppConfigProvider class'.
+   * *
+   * @param {AppConfigGetOptionsInterface} config
+   */
+  public constructor(options: AppConfigGetOptionsInterface) {
+    super();
+    this.client = new AppConfigDataClient(options.clientConfig || {});
+    this.application = options?.sdkOptions?.application || 'app_undefined'; // TODO: make it optional when we add retrieving from env var
+    this.environment = options?.sdkOptions?.environment || 'env_undefined';
+  }
+
+  public async get(name: string, options?: AppConfigGetOptionsInterface): Promise<undefined | string | Uint8Array | Record<string, unknown>> {
+    return super.get(name, options);
+  }
+
+  /**
+   * Retrieve a parameter value from AWS App config.
+   *
+   * @param {string} name - Name of the configuration
+   * @param {AppConfigGetOptionsInterface} options - SDK options to propagate to `StartConfigurationSession` API call
+   * @returns {Promise<Uint8Array | undefined>}
+   */
+  protected async _get(
+    name: string,
+    options?: AppConfigGetOptionsInterface
+  ): Promise<Uint8Array | undefined> {
```

Comment:
Based on [this message](https://github.com/awslabs/aws-lambda-powertools-typescript/pull/1200#issuecomment-1368015600), the return type should include `string` for the empty string case.

Reply:
In the `GetLatestConfigurationResponse` interface `Configuration` may be `Uint8Array` or `undefined`. I didn't find anywhere in the docs that it can return an empty string.